common/src/main/java/com/wynntils/wynn/model/map/MapModel.java

```diff
@@ -67,114 +57,81 @@ public static List<MapTexture> getMapsForBoundingBox(BoundingBox box) {
-                    for (MapPartProfile mapPart : mapPartList) {
-                        String fileName = mapPart.md5 + ".png";
-
-                        handler.addRequest(new RequestBuilder(mapPart.url, "map-part-" + mapPart.name)
-                                .cacheTo(new File(mapDirectory, fileName))
-                                .cacheMD5Validator(mapPart.md5)
-                                .useCacheAsBackup()
-                                .handle(bytes -> {
-                                    try (ByteArrayInputStream in = new ByteArrayInputStream(bytes)) {
-                                        NativeImage nativeImage = NativeImage.read(in);
-                                        MapTexture mapPartImage = new MapTexture(
-                                                fileName, nativeImage, mapPart.x1, mapPart.z1, mapPart.x2, mapPart.z2);
-                                        MAPS.add(mapPartImage);
-                                    } catch (IOException e) {
-                                        WynntilsMod.info(
-                                                "IOException occurred while loading map image of " + mapPart.name);
-                                        return false; // don't cache
-                                    }
-
-                                    return true;
-                                })
-                                .build());
-                    }
-
-                    handler.dispatchAsync();
-                    return true;
-                })
-                .build());
+        Download dl = NetManager.download(UrlId.DATA_STATIC_MAPS);
+        dl.handleReader(reader -> {
+            Type type = new TypeToken<List<MapPartProfile>>() {}.getType();
+
+            List<MapPartProfile> mapPartList = WynntilsMod.GSON.fromJson(reader, type);
+            for (MapPartProfile mapPart : mapPartList) {
+                String fileName = mapPart.md5 + ".png";
+
+                Download dlPart = NetManager.download(URI.create(mapPart.url), "maps/" + fileName, mapPart.md5);
+                dlPart.handleInputStream(
+                        inputStream -> {
+                            try {
+                                NativeImage nativeImage = NativeImage.read(inputStream);
+                                MapTexture mapPartImage = new MapTexture(
+                                        fileName, nativeImage, mapPart.x1, mapPart.z1, mapPart.x2, mapPart.z2);
+                                MAPS.add(mapPartImage);
+                            } catch (IOException e) {
+                                WynntilsMod.info("IOException occurred while loading map image of " + mapPart.name);
+                            }
+                        },
+                        onError -> {
+                            WynntilsMod.info("IOException occurred while loading map image of " + mapPart.name);
```

Comment:
Was this really an IOException? Can we use a different logging message here, so we know where we failed? Also, perhaps log the error itself?